lib/index.js

```diff
@@ -1155,6 +1201,93 @@ class Mcdev {
         Util.logger.info(`:: Done\n`);
         return resultArr;
     }
+    /**
+     * Updates notification email address field
+     *
+     * @param {string} businessUnit name of BU
+     * @param {TYPE.SupportedMetadataTypes} selectedType limit execution to given metadata type
+     * @param {string[]} [keys] customerkey of the metadata
+     * @returns {Promise.<Object.<string, string[]>>} key: business unit name, value: list of affected item keys
+     */
+    static async updateNotifications(businessUnit, selectedType, keys) {
+        let result;
+        if (Util.OPTIONS.clear) {
+            // if no options were provided there is nothing to update
+            result = await this.#runMethod('clearNotifications', businessUnit, selectedType, keys);
+        } else if (
+            !Util.OPTIONS.completionEmail &&
+            !Util.OPTIONS.errorEmail &&
+            !Util.OPTIONS.completionNote &&
+            !Util.OPTIONS.errorNote
+        ) {
+            // if no options were provided there is nothing to update
+            Util.logger.error(`No email addresses or run notes were provided`);
+        } else {
+            // if both clear and notes/emails options were provided
+            // clear will execute first, and then updateNotifications overwriting result
+            result = await this.#runMethod('updateNotifications', businessUnit, selectedType, keys);
+        }
+        return result;
+    }
+    /**
+     * Updates notification email address field
+     *
+     * @param {TYPE.SupportedMetadataTypes} type limit execution to given metadata type
+     * @param {string[]} [keyArr] customerkey of the metadata
+     * @returns {Promise.<Object.<string, string[]>>} key: business unit name, value: list of affected item keys
+     */
+    static async #updateNotificationsGeneric(type, keyArr) {
```

Comment:
That's what I tried to do, but couldn't reuse existing logic that retrieves, updates and deploys importFiles from ImportFile.js(circular dependency error). updateNotifications for importFile does a retrieve, update and then deploy (similar to fixKeys). How do I reuse existing retrieve, and deploy logic in ImportFile and avoid circular dependency error? I tried that at first, but eventually moved this into index.js

For example, to retrieve I would need to import Retriever class. in ImportFile. Retriever uses MetadataTypeInfo, and MetadataTypeInfo imports ImportFile.  So it's ImportFile->Retriever->MetadataTypeInfo->ImportFile

Reply:
as discussed in our call: 
let's reuse `index.js > #fixKeys()` method by making it more generic (similar to runOnBu / runMethod) because updateNotification and fixKeys are kinda doing similar things, checking & updating only specific files during deploy

Reply:
Created a function #updateItems() that does a retrieve and then deploy. Specific options to fixKeys and updateNotifications are set in #fixKeys and #updateNotifications